react/components/Header/components/Modal.js

```diff
@@ -0,0 +1,46 @@
+import { Component } from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+
+let _modalRoot
+
+const getModalRoot = () => {
+  if (typeof _modalRoot === 'undefined') {
+    _modalRoot = document.createElement('div')
+    _modalRoot.classList.add('vtex-modal-root')
+
+    document.body.appendChild(_modalRoot)
+  }
+
+  return _modalRoot
+}
+
+class Modal extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+  }
+
+  constructor(props) {
+    super(props)
+    this.el = document.createElement('div')
+    this.modalRoot = getModalRoot()
```

Comment:
There's no need for this anymore, I found out that you can simply use the `document.body` instead of creating this div.